@here/olp-sdk-dataservice-read/lib/client/VersionedLayerClient.ts

```diff
@@ -36,29 +36,71 @@ import {
     RequestFactory
 } from "..";
 
+export interface VersionedLayerClientParams {
+    // HRN of the catalog.
+    catalogHrn: HRN;
+    // The ID of the layer.
+    layerId: string;
+    // The [[OlpClientSettings]] instance.
+    settings: OlpClientSettings;
+    // Layer version.
+    version: number;
+}
+
 /**
  * Describes a versioned layer and provides the possibility to get partitions metadata and data.
  */
 export class VersionedLayerClient {
     private readonly apiVersion: string = "v1";
-    private readonly hrn: string;
+
+    // HRN of the catalog.
+    private hrn?: string;
+    // The ID of the layer.
+    private layerId?: string;
+    // The [[OlpClientSettings]] instance.
+    private settings?: OlpClientSettings;
+    // Layer version.
+    private version?: number;
 
     /**
+     * @deprecated
      * Creates the [[VersionedLayerClient]] instance.
      *
      * @param catalogHrn The HERE Resource Name (HRN) of the catalog from which you want to get partitions metadata and data.
      * @param layerId The ID of the layer.
      * @param settings The [[OlpClientSettings]] instance.
      * @return The [[VersionedLayerClient]] instance.
      */
-    constructor(
-        catalogHrn: HRN,
-        // The ID of the layer.
-        readonly layerId: string,
-        // The [[OlpClientSettings]] instance.
-        readonly settings: OlpClientSettings
-    ) {
-        this.hrn = catalogHrn.toString();
+    constructor(catalogHrn: HRN, layerId: string, settings: OlpClientSettings);
+
+    /**
+     * Creates the [[VersionedLayerClient]] instance with parameters.
+     * @param params
+     */
+    constructor(params: VersionedLayerClientParams);
+
+    constructor() {
+        const params = arguments;
+
+        console.log("constructor------------");
+        console.log(arguments);
+
+        if (params.length === 3) {
+            this.hrn =
+                params[0] instanceof HRN ? params[0].toString() : undefined;
+            this.layerId = params[1];
+            this.settings = params[2];
+        } else if (params.length === 1) {
+            this.hrn =
+                params[0].catalogHrn && params[0].catalogHrn instanceof HRN
+                    ? params[0].catalogHrn.toString()
+                    : undefined;
+            this.layerId = params[0].layerId;
+            this.settings = params[0].settings;
+            this.version = params[0].version;
+        } else {
+            throw Error("Set correct parameters");
```

Comment:
"Unsupported params, please use VersionedLayerClientParams to initialize client"